linkml_model/model/schema/meta.yaml

```diff
@@ -1420,6 +1420,39 @@ slots:
       - BasicSubset
       - ObjectOrientedProfile
 
+  array_info:
+    domain: slot_definition
+    range: array_info_expression
+    inherited: true
+    description: coerces the value of the slot into an array and defines the dimensions of that array
+    status: testing
+
+  dimensions_info:
+    description: definitions of each axis in the array
+    domain: array_info_expression
+    range: dimension_expression
+    multivalued: true
+    inlined: true
+    status: testing
+
+  minimum_dimensions:
```

Comment:
how do we model mutual exclusivity and constraints between properties here? eg. one shouldn't use `exact_dimensions` alongside `maximum_dimensions`, `maximum_dimensions` shouldn't be less than `minimum_dimensions`, etc. Part of why i like expressing ranges/values in a single object like `dimensions: 3` or `dimensions: {min: 2, max: 3}` is that you can model within the scope of that object, but it seems like that can also be done here i'm just not sure how